_posts/2022-06-22-domain-management-for-multitenant-app-in-aws.markdown

```diff
@@ -0,0 +1,62 @@
+---
+layout: post
+title: Domain Management for Multi-tenant Application in AWS
+description: Architecting multi-tenant applications
+date: 2022-07-03
+author: Shermayne Lee
+image: /img/blog/multitenancy/multitenant.png
+tags: ['multitenant', 'route53', 'DNS', 'DomainManagement']
+---
+
+
+## Introduction
+
+![Domain Management Multitenant](/img/blog/domain-management/multitenantDNS.png)
+
+I have recently started building an application that has to manage multiple clients within the same application. Having few clients doesn't cause any trouble with my implementation initially. Still, when the number of clients grows, it becomes more challenging to keep up with the ongoing tasks of keeping a domain stable and able to support the application. In this post, I would love to share my thoughts on how you can reduce the challenges of managing domains for multiple clients within the same application.
+
+
+Multitenant application has a single instance of an application, and its infrastructure and business logic serves multiple clients. The book [Why Multi-Tenancy is Key to Successful and Sustainable Software-as-a-Service](https://books.apple.com/us/book/why-multi-tenancy-is-key-to-successful-and/id419723802) written by Larry Aiken explains the why it is essential to have multitenancy for some applications as compared to a single tenant.
+
+
+Multitenancy is a challenging problem as you have to consider your application's security, optimization, monitoring, and high availability. Each aspect of implementation could be a real challenge when it grows. For a part like domain management, you don't think it will be a problem as you will only require to do most work during the initial setup. 
+
+However, it can be problematic along the way for a task such as SSL certificate renewal. If you have each certificate for each client's subdomain and each has a different expiration date, you need to manage them manually. Furthermore, for a task like setting up unique subdomains for new clients, what are the measures you take to ensure you will not impact your domains for other clients when there are changes to your current resources. When your number of clients grow, how do you have clear segregation of billing cost for each client? Lastly and most importantly, what is the disaster recovery plan for your instances, and how confident can you bring the application back online with minimal downtime? These are some overhead tasks we need to consider, and often most of the instances mentioned above are handled manually. 
+
+It might have more business value for you to focus on maintaining and developing the application, but managing the domain is critical in keeping the application live and accessible.
```

Comment:
You list out a lot of potential challenges in the previous paragraph and then state here you are focusing on domain but what is the particular particular domain problem you are going to explain how to solve below?  Is it all of them?

Reply:
I would also suggest making this introduction section shorter or putting a one sentence summary of the core issue at the start.  When I am reading posts I want to know in the first couple of lines if it addresses the problem I am trying to solve and if I therefore want to keep reading.